tests/test_parametrized.py

```diff
@@ -90,6 +90,47 @@ def test_unpack_some_modules_static():
     assert sim(x_semantic).shape == torch.Size([n_pix, n_pix])
 
 
+def test_pass_params_as_kwargs():
+    C = FlatLambdaCDM(name="cosmo")
+    lens = Point(cosmology=C)
+    thx, thy = torch.tensor(0.5), torch.tensor(0.5)
+    P = lens.potential(
+        thx,
+        thy,
+        z_s=torch.tensor(1.0),
+        z_l=torch.tensor(0.5),
+        x0=torch.tensor(0.0),
+        y0=torch.tensor(0.0),
+        th_ein=torch.tensor(1.0),
+    )
+    assert torch.all(torch.isfinite(P))
+
+    C.h0 = None
+    with pytest.raises(KeyError):
+        P = lens.potential(
+            thx,
+            thy,
+            z_s=torch.tensor(1.0),
+            z_l=torch.tensor(0.5),
+            x0=torch.tensor(0.0),
+            y0=torch.tensor(0.0),
+            th_ein=torch.tensor(1.0),
+        )
+
```

Comment:
Need to test with batched example

Reply:
I think the `test_batching` test already covers this. You probably wouldn't (couldn't?) vmap with all these kwargs.

Reply:
Yeah that is what I wanted to know, if the kwargs setup could support vmap. I think vmap specifically does not work with kwargs but can work with args.

Reply:
Yeah `vmap` wont work directly with kwargs, but you can write a wrapper if you really want. 
`lambda a,b,c: lens.func(a=a, b=b, c=c)` will turn args to kwargs if its really important. But I think anyone who is vmapping would just use the packed params like this: `lambda x: lens.func(lens.pack(x))` (just added a test for this and it works). Or if they are using a simulator then it packs internally so you can just do `vmap(sim)(x)`, this already has tests.

Reply:
I am a bit confused as to why the pack is necessary in this setup. Can we avoid it and simply have x?